### ISSUE TITLE
NGRX: using addOne is not properly replacing out of date objects

### ISSUE THREAD

**Issue by tylercloke** [chef/automate#1199]
## TLDR

Completely replace our usage of `addOne` in ngrx/store with `upsertOne`.

## Describe the bug

We are currently using `<entityAdapter>.addOne` in the create reducer functions of our NGRX store. Normally, if a user creates an object with an ID that already exists in the store, the API will return an HTTP error code because the object already exists. However, if outside of that specific browser session, that object gets deleted and the user creates the object via the UI, `addOne` will have the undesired behavior of "appending" to the previous object with the same ID in the store instead of overwriting it. Furthermore, when you re-fetch an object, if you don't completely replace the instance of the object, `addOne` will not trigger ngrx/store change detection.

This results in the projects list getting concatenated between the stale object in the store and the newly created, updated, or refetched one. All of our APIs are last write wins so we can trust the response from the POST if it did not result in an error. `<entityAdapter>.upsertOne` appears to have the behavior we want. Let's update all of our create object reducers to use that.

Also should investigate updates as a similar situation might be occurring.

Additional reading:

https://blog.angular-university.io/ngrx-entity/

## To Reproduce

Load the browser with a team with a single project.
Delete that team via the API or a separate web browser / session.
The team should still be in the first browsers store. Create a team with the same ID with different projects. The resulting team should have the original team's projects as well as the new ones (incorrect behavior).

## Expected behavior

Only the projects from the newly created team should be in the store.

**Comment by srenatus** [comment]
Took the liberty of rewording the title. It's not NGRX's fault, it's ours 😉

**Comment by susanev** [comment]
this work has been deprioritized, so closing.
